modules/post/multi/recon/local_exploit_suggester.rb

```diff
@@ -61,7 +70,33 @@ def is_module_platform?(mod)
   end
 
   def is_session_type_compat?(mod)
-    mod.session_compatible? session.sid
+    # These all the `reasons` due to which the module we are checking is not compatible with the current session
+    reasons = mod.session_incompatibility_reasons(session.sid)
+
+    # As we can potentially ignore some `reasons` (e.g. accepting arch values which are, on paper, not compatible),
+    # this keeps track of valid reasons why we will not consider the module that we are evaluating to be valid.
+    valid_reasons = []
+    reasons.each do |reason|
+      # If we are validating e.g. Arch, we want to remove modules that did not pass validation.
+      if !@validate_arch && reason.include?('architecture')
+        vprint_status "#{mod.name} included with override of #{reason}"
+        next
+      end
+      if !@validate_platform && reason.include?('platform')
+        vprint_status "#{mod.name} included with override of #{reason}"
+        next
+      end
+      if !@validate_meterpreter_commands && (reason.start_with?('missing') || reason.start_with?('unloadable'))
+        vprint_status "#{mod.name} included with override of #{reason}"
+        next
+      end
+
+      # If we don't pass verification checks, the reason why a module is not compatible is considered to be `valid`
```

Comment:
I had to read this comment a couple times to understand it, this wording might be more clear or just clearer in my head.
```suggestion
      # If we don't pass verification checks, and this reason is not specifically suppressed it is considered to be `valid`
```